Libs/Widgets/ctkConsole.cpp

```diff
@@ -198,11 +198,60 @@ void ctkConsolePrivate::keyPressEvent(QKeyEvent* e)
       return;
       }
 
-    // Force the cursor back to the interactive area
+    // Paste to the end of document if in the history area
+    if(e == QKeySequence::Paste)
+      {
+      if(history_area)
+        {
+        text_cursor.setPosition(this->documentEnd());
+        this->setTextCursor(text_cursor);  
+        }
+      this->paste();
+      e->accept();
+      return;
+      }
+
+    // Start of line should be the start of interactive area
```

Comment:
Could you consider adding these function to the pimpl: `isKeepAnchorSequence`, `isWordLeftSequence` and `isStartOfLineSequence` ? 

That way the code would be more concise:

```
if(this->isKeepAnchorSequence(e) || this->isWordLeftSequence(e) || this->isStartOfLineSequence(e))
  {
  QTextCursor::MoveMode mode = QTextCursor::MoveAnchor;
  if (this->isKeepAnchorSequence(e))
    {
    mode = QTextCursor::KeepAnchor;
    }
  QTextCursor::MoveOperation op = QTextCursor::Left;
  if (this->isWordLeftSequence(e))
    {
    op = QTextCursor::WordLeft;
    }
  else if (this->isStartOfLineSequence(e))
    {
    op = QTextCursor::StartOfLine;
    }
   [...]
  }
```